internal/pkg/build/sources/verify_sif.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2020, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package sources
+
+import (
+	"context"
+
+	scskeyclient "github.com/sylabs/scs-key-client/client"
+	"github.com/sylabs/singularity/internal/app/singularity"
+	"github.com/sylabs/singularity/pkg/sylog"
+)
+
+// checkSIFFingerprint checks whether a bootstrap SIF image verifies, and was signed with a specified fingerprint
+func checkSIFFingerprint(imagePath string, fingerprints []string, kc *scskeyclient.Config) error {
+	sylog.Infof("Checking bootstrap image verifies with fingerprint(s): %v", fingerprints)
+	opts := []singularity.VerifyOpt{}
+	if kc != nil {
+		opts = append(opts, singularity.OptVerifyUseKeyServer(kc))
+	}
+	ctx := context.TODO()
+	return singularity.VerifyFingerprints(ctx, imagePath, fingerprints, opts...)
+}
+
+// verifySIF checks whether a bootstrap SIF image verifies
+func verifySIF(imagePath string, kc *scskeyclient.Config) error {
+	sylog.Infof("Verifying bootstrap image %s", imagePath)
+	opts := []singularity.VerifyOpt{}
+	if kc != nil {
+		opts = append(opts, singularity.OptVerifyUseKeyServer(kc))
+	}
+	ctx := context.TODO()
+	return singularity.Verify(ctx, imagePath, opts...)
+}
```

Comment:
Might be worth opening a tech debt issue to addd contexts here? These are both using a key server some of the time, which is potentially a long-running operation.

Reply:
Agreed - I'll put a // TODO comment and open issues. I think we quite a way below where we have a context... but we did some work in other area which would be good to reprise here.

Reply:
On second look - I'll fix this properly.... it's not too far away.